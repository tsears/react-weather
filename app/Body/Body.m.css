.body {
  flex-grow: 1;
  padding: 2rem;
  min-height: 100%;
}

.heading {
  margin-bottom: 1rem;
}
