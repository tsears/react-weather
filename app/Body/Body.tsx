import * as React from 'react'
import * as styles from './Body.m.css'

interface BodyState { lmao: string }

export class Body extends React.Component<{}, BodyState> {
  public constructor (props: {}) {
    super(props)

    this.state = {
      lmao: null,
    }
  }

  public render (): React.ReactElement {
    const { lmao } = this.state

    return (
      <div className={styles.body}>
        <h1 className={styles.heading}>React Scaffold</h1>
        <div>Ayy {lmao}</div>
      </div>
    )
  }

  public async componentDidMount (): Promise<void> {
    const response = await fetch('/api/ayy')
    if (!response.ok) {
      throw new Error('HTTP error, status = ' + response.status)
    }

    const data = await response.json()

    this.setState({
      lmao: data.response,
    })
  }
}
