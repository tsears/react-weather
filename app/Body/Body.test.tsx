import * as React from 'react'
import { create, act, ReactTestRenderer } from 'react-test-renderer'
import { Body } from './Body'

global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve({ response: 'test' }),
    ok: true,
    status: 200,
  })
) as jest.Mock

Object.defineProperty(global.navigator, 'geolocation', {
  value: {
    getCurrentPosition: jest.fn(),
  },
})

beforeEach(() => { (fetch as jest.Mock).mockClear() })

describe('Body Component', () => {
  it('Renders correctly', async () => {
    let tree: ReactTestRenderer

    await act(() => {
      tree = create(<Body weatherData={undefined} locationData={undefined}/>)
    })

    expect((tree).toJSON()).toMatchSnapshot()
  })
})
