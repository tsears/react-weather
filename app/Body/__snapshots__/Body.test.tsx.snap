// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body Component Renders correctly 1`] = `
<div>
  <div>
    <div>
      <div>
        <div>
          <div />
          <div />
          <div />
          <div />
        </div>
      </div>
    </div>
  </div>
</div>
`;
